lib/fluent/plugin/formatter_hash.rb

```diff
@@ -22,10 +22,21 @@ class HashFormatter < Formatter
       Plugin.register_formatter('hash', self)
 
       config_param :add_newline, :bool, default: true
+      config_param :newline, :enum, list: [:lf, :crlf], default: :lf
```

Comment:
How about using `Fluent.windows?` for changing default for windows?

In addition, `newline` code are duplicated in many places. How about providing mixin in Formatter?

Reply:
I'd applied the above suggestion as https://github.com/fluent/fluentd/pull/3152/commits/0943b1436a2882c5d0980c223b34e8f374661ecc and https://github.com/fluent/fluentd/pull/3152/commits/fdcfaf4ec0c5abd4a28c535e577541305b60f0e7.